OpenDreamRuntime/Objects/DreamIcon.cs

```diff
@@ -83,6 +83,9 @@ public sealed class IconFrame {
         if (_cachedDMI != null)
             return _cachedDMI;
 
+        if(Width == 0 && Height == 0)
+           Width = Height = 32; //TODO should be world.icon_size
```

Comment:
Should this instead set the size of `Width` or `Height` if either of them is 0, instead of only when both are?